### ISSUE TITLE
#13354 triggers tons of false positives when unable to resolve output type

### ISSUE THREAD

**Issue by Ten0** [rust-lang/rust-analyzer#13430]
I'm getting tons of false positives since today's release of #13354:
```
the return type of the containing function does not implement `FromResidual`
```
(incorrect-try-target)

This happens when a function is returning a `Result<_, Fail<T>>` and rust-analyzer is not able to infer the concrete type `T`.

I suspect that is specifically [here](https://github.com/rust-lang/rust-analyzer/pull/13354/files#diff-29a95b03c52f874531156aa9fea08370b25ec9c16a75bb40aa5bfbe3d977265dR568-R589) that something is incorrect, but I'm not expert enough on RA to be able to read the meaning of all of the `Option` layers there proficiently. I suspect someone might be able to identify why this happens by just reading my above description - please let me know if more details are needed.

(I failed to create a minimal repro because I'm not sure how specifically to build a code in which RA is unable to infer the return type, although these are pretty common in Diesel)

**rust-analyzer version**: 0.3.1248-standalone
**rustc version**: rustc 1.63.0

NB: disabling the diagnostic makes RA usable again:
```json
	"rust-analyzer.diagnostics.disabled": [
		"incorrect-try-target"
	],
```

**Comment by lnicola** [comment]
Dupe of #13429

**Comment by Ten0** [comment]
When you take time to write the issue and by the time you submit it somebody else has already submitted it... you know it's gonna get fixed quickly ^^

**Comment by Veykril** [comment]
What exactly is the `_` in `Result<_, Fail<T>>`?

**Comment by Ten0** [comment]
> What exactly is the `_` in `Result<_, Fail<T>>`?

Whatever it is the issue happens as far as I can tell.
